src/test/java/vvs_dbsetup/CustomersDBTest.java

```diff
@@ -0,0 +1,127 @@
+import org.junit.*;
+
+import com.ninja_squad.dbsetup.DbSetup;
+import com.ninja_squad.dbsetup.DbSetupTracker;
+import com.ninja_squad.dbsetup.Operations;
+import com.ninja_squad.dbsetup.destination.Destination;
+import com.ninja_squad.dbsetup.destination.DriverManagerDestination;
+import com.ninja_squad.dbsetup.operation.Operation;
+
+import static vvs_dbsetup.DBSetupUtils.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import webapp.persistence.PersistenceException;
+import webapp.services.*;
+
+/**
+ * Tests the expected behavior of Customers interactions with the database
+ * @author fc45701
+ * @author fc52214
+ */		
+public class CustomersDBTest {
+
+	private static Destination dataSource;
+	
+    // the tracker is static because JUnit uses a separate Test instance for every test method.
+    private static DbSetupTracker dbSetupTracker = new DbSetupTracker();
+	
+    @BeforeClass
+    public static void setupClass() {
+    	startApplicationDatabaseForTesting();
+		dataSource = DriverManagerDestination.with(DB_URL, DB_USERNAME, DB_PASSWORD);
+    }
+    
+	@Before
+	public void setup() throws SQLException {
+
+		Operation initDBOperations = Operations.sequenceOf(
+			  DELETE_ALL
+			, INSERT_CUSTOMER_SALE_DATA
+			);
+		
+		DbSetup dbSetup = new DbSetup(dataSource, initDBOperations);
+		
+        // Use the tracker to launch the DbSetup. This will speed-up tests 
+		// that do not not change the BD. Otherwise, just use dbSetup.launch();
+        dbSetupTracker.launchIfNecessary(dbSetup);
+		
+	}
+	
+	/**
+	 * a) The SUT does not allow to add a new client with an existing VAT
+	 */
+	@Test
+	public void noDuplicateVATTest() throws ApplicationException {
+		int vat = 197672337;
+		assumeTrue(hasClient(vat));
+		assertThrows(Exception.class, () -> 
+			CustomerService.INSTANCE.addCustomer(vat, "FCUL", 217500000));
+	}
+	/**
+	 * b) after the update of a costumer contact, 
+	 * 	  the information should be properly saved
+	 */
+	@Test
+	public void updatingContactSavesTheChangesTest() throws ApplicationException {
+		int vat = 197672337;
+		CustomerDTO cust = CustomerService.INSTANCE.getCustomerByVat(vat);
+		assumeTrue(hasClient(vat));
+		CustomerService.INSTANCE.updateCustomerPhone(vat, cust.phoneNumber + 1);
+		assertTrue(CustomerService.INSTANCE.getCustomerByVat(vat).phoneNumber == cust.phoneNumber + 1);
+	}
+	
+	/**
+	 * c) after deleting all costumers, the list of all customers should be empty
+	 */
+
+	@Test
+	public void deletingAllCustomersTest() throws ApplicationException {
+		assertTrue(CustomerService.INSTANCE.getAllCustomers().customers.size() > 0);
+		deleteAllCostumers();
+		assertTrue(CustomerService.INSTANCE.getAllCustomers().customers.size() == 0);
+	}
+
+	/**
+	 * d) after deleting a certain costumer,
+	 *    it's possible to add ir back without lifting exceptions
+	 */
+	@Test
+	public void addingDeletedCostumerTest() throws ApplicationException {
+		int vat = 197672337;
+		assumeTrue(hasClient(vat));
```

Comment:
if this vat is not in the db, the test doesn't do anything and passes (with a warning?). I think it would be good to take the number from the db (sth like getAllCustomers().get(0) )